File: External-Tutorial-Dagger-Android-Sample/src/main/java/com/gabrielsamojlo/dagger_212/DummyDependency.java
package com.gabrielsamojlo.dagger_212;

import android.content.Context;

public class DummyDependency {

    private Context context;

    public DummyDependency(Context context) {
        this.context = context;
    }

    public String getApplicationName() {
        return context.getString(R.string.app_name);
    }

}

File: Tutorial7-3BindsContextWithApplication/src/main/java/com/smarttoolfactory/tutorial7_3bindscontextwithapplication/model/DummyDependency.kt
package com.smarttoolfactory.tutorial7_3bindscontextwithapplication.model

import android.content.Context
import com.smarttoolfactory.tutorial7_3bindscontextwithapplication.R

class DummyDependency(private val context: Context) {

    val applicationName: String
        get() = context.getString(R.string.app_name)

}